Frontend/student-hub/src/components/Navbar.js

```diff
@@ -37,65 +36,71 @@ function Navbar({ isLoggedIn, setIsLoggedIn}) {
                             edge="end"
                             color="inherit"
-                            onClick={handleSettingsClick}
+                            onClick={handleProfileClick}
                             >
-                                <SettingsIcon />
+                                <AccountCircle />
                             </IconButton>
                             <Menu
-                            anchorEl={settingsEl}
-                            keepMounted
-                            open={Boolean(settingsEl)}
-                            onClose={handleSettingsClose}
-                            >
-                                <MenuItem onClick={() => { navigate('/feedback'); handleSettingsClose(); }}>Feedback</MenuItem>
-                                <MenuItem onClick={() => { navigate('/contact'); handleSettingsClose(); }}>Contact Me</MenuItem>
-                                <MenuItem onClick={() => { navigate('/about'); handleSettingsClose(); }}>About</MenuItem>
-                            </Menu>
-                    </>
-                ) : (
-                    undefined
-                )}
-            </Toolbar>
-        </AppBar>
-    );
-}
+                                anchorEl={anchorEl}
+                                keepMounted
+                                open={Boolean(anchorEl)}
+                                onClose={handleClose}
+                                >
+                                    <MenuItem onClick={() => { navigate('/profile'); handleClose(); }}>Profile</MenuItem>
+                                    <MenuItem onClick={handleLogout}>Logout</MenuItem>
+                                </Menu>
+                                <IconButton
+                                edge="end"
+                                color="inherit"
+                                onClick={handleSettingsClick}
+                                >
+                                    <SettingsIcon />
+                                </IconButton>
+                                <Menu
+                                anchorEl={settingsEl}
+                                keepMounted
+                                open={Boolean(settingsEl)}
+                                onClose={handleSettingsClose}
+                                >
+                                    <MenuItem>
+                                        <Switch checked={darkMode} onChange={toggleDarkMode} />
+                                        Dark Mode
+                                    </MenuItem>
+                                    <MenuItem onClick={() => { navigate('/feedback'); handleSettingsClose(); }}>Feedback</MenuItem>
+                                    <MenuItem onClick={() => { navigate('/contact'); handleSettingsClose(); }}>Contact Me</MenuItem>
+                                    <MenuItem onClick={() => { navigate('/about'); handleSettingsClose(); }}>About</MenuItem>
```

Comment:
there's some repetitive code in each of the `onClick`s here - let's try to see if we can create a reusable function and just pass in the path as a parameter!